rwengine/src/audio/SoundSource.cpp

```diff
@@ -0,0 +1,487 @@
+                        0, nullptr);
+                    if (!swr) {
+                        RW_ERROR(
+                            "Resampler has not been successfully allocated.");
+                        return;
+                    }
+                    swr_init(swr);
+                    if (!swr_is_initialized(swr)) {
+                        RW_ERROR(
+                            "Resampler has not been properly initialized.");
+                        return;
+                    }
+                }
+
+                // Decode audio packet
+                if (receiveFrame == 0 && sendPacket == 0) {
+                    // Write samples to audio buffer
+                    resampled = av_frame_alloc();
+                    resampled->channel_layout = AV_CH_LAYOUT_STEREO;
+                    resampled->sample_rate = frame->sample_rate;
+                    resampled->format = kOutputFMT;
+                    resampled->channels = kNumOutputChannels;
+
+                    swr_config_frame(swr, resampled, frame);
+
+                    if (swr_convert_frame(swr, resampled, frame) < 0) {
+                        RW_ERROR("Error resampling " << filePath << '\n');
+                    }
+
+                    for (size_t i = 0;
+                         i <
+                         static_cast<size_t>(resampled->nb_samples) * channels;
+                         i++) {
+                        data.push_back(
+                            reinterpret_cast<int16_t*>(resampled->data[0])[i]);
+                    }
+                    av_frame_unref(resampled);
+                }
+            }
+        }
+    }
+
+#endif
+
+    // Cleanup
+    /// Free all data used by the frame.
+    av_frame_free(&frame);
+
+    /// Free resampler
+    swr_free(&swr);
+
+    /// Close the context and free all data associated to it, but not the
+    /// context itself.
+    avcodec_close(codecContext);
+
+    /// Free the context itself.
+    avcodec_free_context(&codecContext);
+
+    /// We are done here. Close the input.
+    avformat_close_input(&formatContext);
+}
+
+/// Structure for input data
+struct InputData {
+    uint8_t* ptr = nullptr;
+    size_t size{};  ///< size left in the buffer
+};
+
+/// Low level function for copying data from handler (opaque)
+/// to buffer.
+static int read_packet(void* opaque, uint8_t* buf, int buf_size) {
+    auto* input = reinterpret_cast<InputData*>(opaque);
+    buf_size = FFMIN(buf_size, input->size);
+    /* copy internal data to buf */
+    memcpy(buf, input->ptr, buf_size);
+    input->ptr += buf_size;
+    input->size -= buf_size;
+    return buf_size;
+}
+
+void SoundSource::loadSfx(const rwfs::path& path, LoaderSDT& sdt, size_t index, bool asWave) {
+    // Allocate audio frame
+    AVFrame* frame = av_frame_alloc();
+    if (!frame) {
+        RW_ERROR("Error allocating the audio frame");
+        return;
+    }
+
+
+    sdt.load(path / "audio/sfx");
```

Comment:
It should be done only once in ctor of SoundManager. @madebr can/should we keep path in SoundSource?

Reply:
To do later.